issues with globus-sdk 4.8.0

[globus-sdk 4.8.0](https://globus-sdk-python.readthedocs.io/en/stable/changelog.html) causes some issues with unit-tests on [iblrig](https://github.com/int-brain-lab/iblrig) related to their [added support for orjson](https://github.com/globus/globus-sdk-python/pull/1385):

```
FAILED iblrig/test/test_misc.py::TestPortSettings::test_port_settings - RuntimeError: 'orjson' is not available but globus-sdk was configured to use it. This is not valid. Please ensure that 'orjson' is installed.
FAILED iblrig/test/test_video.py::TestCameraSession::test_run_video_session - RuntimeError: 'orjson' is not available but globus-sdk was configured to use it. This is not valid. Please ensure that 'orjson' is installed.
```

This seems to be an upstream issue with globus-sdk. Options:

- fix upstream issue,
- add orjson dependency (might be nice to have for performance reasons), or
- pin globus-sdk to < 4.8.0

--- a/changelog.d/20260616_111455_kurtmckee_fix_orjson_compat_getattr.rst
+++ b/changelog.d/20260616_111455_kurtmckee_fix_orjson_compat_getattr.rst
@@ -1,0 +1,7 @@
+Fixed
+-----
+
+- Fix a crash bug in the orjson compatibility module. (:pr:`NUMBER`)
+
+  This manifested as a ``RuntimeError``, and could occur when a test framework
+  accessed the module to check for warning registries in the module.

--- a/src/globus_sdk/_internal/orjson_compat.py
+++ b/src/globus_sdk/_internal/orjson_compat.py
@@ -35,10 +35,10 @@ else:
         return ["__all__", "__file__", "__path__"] + list(__all__)
 
     def __getattr__(name: str) -> t.Any:
-        require()
-
-        mod = sys.modules[__name__]
         if name in ("loads", "dumps"):
+            require()
+
+            mod = sys.modules[__name__]
             value = getattr(orjson, name)
             setattr(mod, name, value)
             return value

